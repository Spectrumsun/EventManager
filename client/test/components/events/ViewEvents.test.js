import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import ConnectedViewEvent,
{
  ViewEvent,
  mapDispatchToProps } from '../../../src/components/Event/ViewEvents';
import Display from '../../../src/components/Event/getEvent'


const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  events: {
    events: [{
      id: 10,
      eventName: 'tom',
      startDate: '2018-10-12',
      endDate: '2018-10-15',
      purpose: 'test',
      centers: {
        id: 100,
        imageurl: 'https:image.com/gif'
      }
    }],
    error: false
  }
};

const store = mockStore(initialState);

const props = {
  onInitEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  error: false,
  history: createMemoryHistory(),
  events: [{
    id: 10,
    eventName: 'tom',
    startDate: '2018-10-12',
    endDate: '2018-10-15',
    purpose: 'test',
    centers: {
      id: 101,
      imageurl: 'https:image.com/gif'
    }
  }],
};

describe('<ViewEvent /> Component', () => {
  const wrapper = shallow(<ViewEvent {...props} />);
  it('should render the <ViewEvent />', () => {
    shallow(<ViewEvent {...props} />);
  });


  it('should match component snapshot', () => {
    const tree = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedViewEvent {...props} />
        </BrowserRouter>
      </Provider>);
    expect(tree).toMatchSnapshot();
  });

  it('should render initial layout of <ViewEvent />', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should have div on layout', () => {
    expect(wrapper.find('div').length).toEqual(2);
  });

  it('should have <Event /> when the page loads', () => {
    expect(wrapper.find(Display)).toHaveLength(1);
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).onInitEvent).toBeTruthy();
  });
});

