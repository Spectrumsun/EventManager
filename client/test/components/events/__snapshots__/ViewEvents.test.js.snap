// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewEvent /> Component should match component snapshot 1`] = `
<div>
  <div
    className="center thebody"
  >
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Events
    </h1>
    <a
      href="/events/10"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
      <div
        className="card d-lg-inline-block"
        style={
          Object {
            "width": "20rem",
          }
        }
      >
        <img
          alt="Card cap"
          className="card-img-top"
          src="https:image.com/gif"
          style={
            Object {
              "height": "15rem",
              "objectFit": "cover",
            }
          }
        />
        <div
          className="card-body"
        >
          <h4
            className="card-title"
          >
            tom
          </h4>
          <span
            className="card-text"
          >
            2018-10-12
          </span>
           To 
          <span
            className="card-text"
          >
            2018-10-15
          </span>
          <h6
            className="card-text"
          >
            test
          </h6>
          <h6
            className="btn btn-primary"
          >
            View more
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="center"
  >
    <footer>
      <div>
        <h3
          className="center"
        >
          Event Manager
        </h3>
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
            "textAlign": "left",
          }
        }
      >
        <div>
          <div
            className="float-left"
          >
            <a
              className="navbar-brand"
              href="/"
            >
              <img
                alt=""
                height="30"
                src="https://res.cloudinary.com/skybound/image/upload/v1522444986/eventmanager/static/event.png"
                width="30"
              />
            </a>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Social Media
              </p>
              <li
                className="footerColor"
              >
                facebook
              </li>
              <li
                className="footerColor"
              >
                twitter
              </li>
              <li
                className="footerColor"
              >
                instargram
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Event manager
              </p>
              <li
                className="footerColor"
              >
                Questions
              </li>
              <li
                className="footerColor"
              >
                Jobs
              </li>
              <li
                className="footerColor"
              >
                Help
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                About
              </p>
              <li
                className="footerColor"
              >
                desgin by Taiwo Sunday
              </li>
              <li
                className="footerColor"
              >
                +2348038512455
              </li>
              <li
                className="footerColor"
              >
                https://github.com/Spectrumsun
              </li>
              <li
                className="footerColor"
              >
                taiwo.sunday@outlook.com
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Site
              </p>
              <li
                className="footerColor"
              >
                Home
              </li>
              <li
                className="footerColor"
              >
                Centers
              </li>
              <li
                className="footerColor"
              >
                Events
              </li>
            </ul>
          </div>
        </div>
      </div>
      <p
        className="center"
      >
        Event Managment Â© 2017
      </p>
    </footer>
  </div>
</div>
`;

exports[`<ViewEvent /> Component should render initial layout of <ViewEvent /> 1`] = `
Array [
  <div>
    <div
      className="center thebody"
    >
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Events
      </h1>
      <Link
        replace={false}
        style={
          Object {
            "color": "black",
          }
        }
        to="/events/10"
      >
        <display
          centerImage="https:image.com/gif"
          endDate="2018-10-15"
          eventName="tom"
          purpose="test"
          startDate="2018-10-12"
        />
      </Link>
    </div>
    <footer />
  </div>,
]
`;
